chain/chain/src/chain.rs

```diff
@@ -2376,15 +2380,15 @@ impl Chain {
 
     fn get_catchup_and_state_sync_infos(
         &self,
-        header: &BlockHeader,
-        prev_hash: CryptoHash,
-        prev_prev_hash: CryptoHash,
+        epoch_id: &EpochId,
+        block_hash: &CryptoHash,
+        prev_hash: &CryptoHash,
+        prev_prev_hash: &CryptoHash,
         me: &Option<AccountId>,
-        block: &MaybeValidated<Block>,
     ) -> Result<(bool, Option<StateSyncInfo>), Error> {
-        if self.epoch_manager.is_next_block_epoch_start(&prev_hash)? {
-            debug!(target: "chain", block_hash=?header.hash(), "block is the first block of an epoch");
-            if !self.prev_block_is_caught_up(&prev_prev_hash, &prev_hash)? {
+        if self.epoch_manager.is_next_block_epoch_start(prev_hash)? {
+            debug!(target: "chain", %block_hash, "block is the first block of an epoch");
```

Comment:
mini nit: The log is not wrong but the message is a bit out of place. It's very generic but it's logged from within `get_catchup_and_state_sync_infos`. Maybe log something like `get_catchup_and_state_sync_infos`

Reply:
moved it to the bottom to log what shards were gonna state sync